### PR TITLE
Filters as url params

### PR DESCRIPTION
**Pull request by mdtrooper**
I added some features to the page:

* when it is filtered by tag or filter, it is stored as url params.
* added a count of games filtered in the top menu, 100/1019 for example.

Regards.

### CODE REVIEW

**Comment by cxong** [comment]
oh that's a nice idea! Will test this out soon

**Comment by mdtrooper** [comment]
> oh that's a nice idea! Will test this out soon

Yes I think it is good. I added a screenshot with filtered games per "train" topic:

image

File: index.html [hunk 1]
```diff
@@ -46,6 +46,9 @@
             </div>
           </div>
         </li>
+        <li>
+          <span class="nav-count">xxxx/yyyy games</span>
```
Comment (cxong) [inline index.html]: 👍

File: static/main.js [hunk 2]
```diff
@@ -42,42 +44,78 @@ var OSGC = window.OSGC = {};
...
-            index.push(next.getAttribute('data-index'));
-        }
-        el.setAttribute('data-index', index.join(' '));
-    };
+function getfilter(term) {
+  return !term ? "" :
+    '[data-index*="' + term.toLowerCase().replace('"', '') + '"]';
+}
+
+var filter_style = document.getElementById('filter-style');
```
Comment (cxong) [inline static/main.js]: use camelCase `filterStyle`

File: static/main.js [hunk 3]
```diff
@@ -42,42 +44,78 @@ var OSGC = window.OSGC = {};
...
-    for (var i = 0, l = nodes.length; i < l; i++) {
-        var el = nodes[i], next = el, index = [];
-        while ((next = next.nextElementSibling) && !next.id) {
-            if (next.tagName != 'DD')
-                continue;
-            index.push(next.getAttribute('data-index'));
-        }
-        el.setAttribute('data-index', index.join(' '));
-    };
+function getfilter(term) {
```
Comment (cxong) [inline static/main.js]:
```suggestion
function getFilter(term) {
```

File: static/main.js [hunk 4]
```diff
@@ -42,42 +44,78 @@ var OSGC = window.OSGC = {};
...
-            "{display: block}";
-    });
+  document.getElementById('filter').addEventListener('input', function() {
+    filter(this.value);
+  });
 })();
 
 // tag handling
-(function() {
+function filterByTag(curTag) {
```
Comment (cxong) [inline static/main.js]: 👍

File: static/main.js [hunk 5]
```diff
@@ -42,42 +44,78 @@ var OSGC = window.OSGC = {};
...
-        el.setAttribute('data-index', index.join(' '));
-    };
+function getFilter(term) {
+  return !term ? "" :
+    '[data-index*="' + term.toLowerCase().replace('"', '') + '"]';
+}
+
+var filterStyle = document.getElementById('filter-style');
+
+function filter(filter_value) {
```
Comment (cxong) [inline static/main.js]:
```suggestion
function filter(filterValue) {
```

File: static/main.js [hunk 6]
```diff
@@ -279,3 +296,72 @@ var OSGC = window.OSGC = {};
...
+      value = isNaN(Number(value))? value : Number(value);
+      return (q[key] = value, q);
+    }, {});
+    
+    return queryParams;
+}
+
+function setQueryParams(key, value) {
+  if (value === null) {
+    params = Object.keys(params).reduce(function(object, key_it) {
```
Comment (cxong) [inline static/main.js]:
```suggestion
    params = Object.keys(params).reduce(function(object, keyIt) {
```

File: static/main.js [hunk 7]
```diff
@@ -279,3 +296,72 @@ var OSGC = window.OSGC = {};
...
+    }
+    url += encodeURIComponent(key) + '=' + encodeURIComponent(params[key]);
+  });
+  history.replaceState({}, null, url);
+}
+
+function setCount() {
+  var count = ' games';
+  var total = document.getElementsByTagName('dd').length;
+  var hidden = Array.prototype.slice.call(document.getElementsByTagName('dd')).reduce(function(count_hidden, game) {
```
Comment (cxong) [inline static/main.js]:
```suggestion
  var hidden = Array.prototype.slice.call(document.getElementsByTagName('dd')).reduce(function(countHidden, game) {
```

**Comment by mdtrooper** [comment]
The changes is done. Thanks for the suggestions.

**Pull request merged**